assets/javascripts/discourse/components/chat-live-pane.js

```diff
@@ -196,7 +197,7 @@ export default Component.extend({
   },
 
   loadDraftForChannel(channelId) {
-    this.set("draft", this.chat.getDraftForChannel(channelId));
+    this.set("draft", this.chatDraftHandler.getDraftForChannel(channelId));
```

Comment:
Do we need to create a local property here? Can't we just always the draft from the handler?

Reply:
Currently I've created a computed prop like the following.

```
  @discourseComputed("chatChannel.id")
  draft(channelId) {
    return this.chatDraftHandler.getForChannel(channelId);
  }
```

But its not working. I'm investigating why.

Reply:
Thanks for the pointer, its really helpful. But in this case, its an old school factory component.

What actually fixes it though, is if I make it a local property as it was earlier and set it it in `didReceiveAttrs`

Reply:
Sorry was on mobile, I didn't have context, and usually this is the issue :p

Reply:
I've made it a local property. The probably reason for computed prop not working is, it needs to be set at a point in time but computed is set as soon as its read the first time.

I can dig further if needed.